Basic-Recruit-Training/Person.cpp

```diff
@@ -0,0 +1,73 @@
+#include <iostream> //Imports cout
+#include <sstream>
+#include <string>
+#include "Person.h"
+
+// Person::Person();  //creating what the functions actually DO here
+Person::Person()    {
+name_ = "";
+age_ = new int;
+
+}
+
+
+ Person::Person(const std::string& name, int a)   {
+ name_ = name;
+// a = new int; //do I have to create this block of memory, or is it unessesary? ask Bill
+ age_ = new int(a);
+
+ }
+
+ Person::~Person()  {
+    delete age_;
+ }
+
+ int Person::getAge() const {
+    return *age_;
+ }
+const std::string& Person::getName() const  {
+return name_;
+
+}
+
+void Person::printInfo() const   {
+ std::cout << "Name: " << getName() << std::endl
+              << "Age: " << getAge() << std::endl;
+
+}
+//void Person::printInfo() const  {
+//stf::count << Bob.getName() << std::endl << Bob.getAge() << std::endl;
+//}
+
+
+void Person::setName(const std::string& newName)  {
+    name_ = newName;
+}
+void Person::setAge(int newAge)   {
+    delete(age_);   //freeing this memory space in the heap
```

Comment:
Here delete is not necessary, we have already allocated memory for age, so all we need to do is just change the value